src/ai/defaultai.cc

```diff
@@ -1478,23 +1478,30 @@ void DefaultAI::update_all_mineable_fields(const Time& gametime) {
 }
 
 /**
- * Checks up to 50 fields that weren't buildable the last time.
- *
- * milliseconds if the area the computer owns is big.
+ * Checks a part of unusable_fields. Outcome can be:
+ * - the field is not ours anymore - drop it
+ * - it is still ours, but still not buildable - do nothing (keep it)
+ * - is ours, and buildable, drop from unused fields and create buildable_field or
+ * mineable_field and insert to particular dequeue
  */
 void DefaultAI::update_all_not_buildable_fields() {
 	int32_t const pn = player_number();
 
 	// We are checking at least 5 unusable fields (or less if there are not 5 of them)
-	// at once, but not more then 200...
+	// at once, but not more then 400...
 	// The idea is to check each field at least once a minute, of course with big maps
 	// it will take longer
 	uint32_t maxchecks = unusable_fields.size();
 	if (maxchecks > 5) {
-		maxchecks = std::min<uint32_t>(5 + (unusable_fields.size() - 5) / 15, 200);
+		maxchecks = std::min<uint32_t>(5 + (unusable_fields.size() - 5) / 10, 400);
 	}
 
-	for (uint32_t i = 0; i < maxchecks; ++i) {
+	// for performance reasons we update only this count of fields
+	const uint32_t fields_update_limit = 5;
```

Comment:
did you play with the numbers? 5 seems to be very few if 400 are pushed to the other deques

how would 20 work on your machine

Reply:
The reasoning for low number is that update_all_buildable_fields() method will have a sophisticated logic for picking fields (as just discussed in the other pull request) so I did not want to go too high here. But 20 is fine too..